spec/jobs/resync_document_job_spec.rb

```diff
@@ -0,0 +1,17 @@
+RSpec.describe ResyncDocumentJob do
+  include ActiveJob::TestHelper
+
+  before do
+    populate_default_government_bulk_data
+  end
+
+  it "retries the job when an exception is raised" do
+    error = GdsApi::TimedOutException.new
+    stub_any_publishing_api_call.to_raise(error)
+
+    document = create(:document, :with_current_and_live_editions)
+    described_class.perform_now(document)
+
+    expect(described_class).to have_been_enqueued
```

Comment:
You can simplify this a bit and not need to create extra editions if you modify this to be something like:

```
    allow(ResyncDocumentService).to receive(:call).and_raise(GdsApi::BaseError)
    described_class.perform_now(build(:document))
    expect(described_class).to have_been_enqueued
```

Reply:
I have to use `create` rather than `build` otherwise I get the following error: 

```
1) ResyncDocumentJob retries the job when an exception is raised
     Failure/Error: described_class.perform_now(build(:document))
     
     ActiveJob::SerializationError:
       Unable to serialize Document without an id. (Maybe you forgot to call save?)
     # ./spec/jobs/resync_document_job_spec.rb:8:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # GdsApi::BaseError:
     #   GdsApi::BaseError
     #   ./app/jobs/resync_document_job.rb:9:in `perform'
```